lighthouse-core/audits/seo/font-size.js

```diff
@@ -0,0 +1,233 @@
+/**
+ * @license Copyright 2017 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const parseURL = require('url').parse;
+const Audit = require('../audit');
+const ViewportAudit = require('../viewport');
+const CSSStyleDeclaration = require('../../lib/web-inspector').CSSStyleDeclaration;
+const MINIMAL_PERCENTAGE_OF_LEGIBLE_TEXT = 75;
+
+/**
+ * @param {Array<{cssRule: WebInspector.CSSStyleDeclaration, fontSize: number, textLength: number, node: Node}>} fontSizeArtifact
+ * @returns {Array<{cssRule: WebInspector.CSSStyleDeclaration, fontSize: number, textLength: number, node: Node}>}
+ */
+function getUniqueFailingRules(fontSizeArtifact) {
+  const failingRules = new Map();
+
+  fontSizeArtifact.forEach(({cssRule, fontSize, textLength, node}) => {
+    const artifactId = getFontArtifactId(cssRule, node);
```

Comment:
perhaps something got thrown off between the iterations but it seems like analyzed text nodes aren't always getting de-duped anymore?

![image](https://user-images.githubusercontent.com/2301202/33564495-7e81f9ce-d8cf-11e7-8cd0-e56cc76629dd.png)

rule is in CSS
![image](https://user-images.githubusercontent.com/2301202/33564523-8fd959e2-d8cf-11e7-9c41-73e30a051eb1.png)

Reply:
ah ok, great catch!